tests/kafkatest/tests/core/replication_test.py

```diff
@@ -138,10 +138,11 @@ def min_cluster_size(self):
     @parametrize(failure_mode="hard_bounce",
             broker_type="leader",
             security_protocol="SASL_SSL", client_sasl_mechanism="SCRAM-SHA-256", interbroker_sasl_mechanism="SCRAM-SHA-512")
-    def test_replication_with_broker_failure(self, failure_mode, security_protocol,
-                                             broker_type, client_sasl_mechanism="GSSAPI",
-                                             interbroker_sasl_mechanism="GSSAPI",
-                                             enable_idempotence=False):
+    @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
+            security_protocol=["PLAINTEXT"], broker_type=["leader"], compression_type=["gzip"])
```

Comment:
I'm a little concerned that it's getting more and more difficult to tell whether all the interesting cases are covered. For example, I was wondering if we should be testing the compression path with idempotence enabled as well (maybe not?). Since replication is such a critical part of Kafka, I guess I'd rather err on the side of too much coverage even if we end up with some redundancy which causes longer test times. If this test is taking too long, maybe we can open an issue to investigate whether all of that time is well spent.